scripts/generate_leaderboard.py

```
@@ -51,11 +72,15 @@ def fetch_search_pulls(org: str, state: str, page: int):
             if remaining == "0":
                 raise RuntimeError("GitHub API rate limit reached. Please try later.")
             raise RuntimeError("GitHub API access is temporarily restricted (403).") from error
+        if error.code == 422:
+            raise RuntimeError(
+                "GitHub search query exceeded API limits. Narrow the date range."
+            ) from error
         raise RuntimeError(f"GitHub API error: {error.code}") from error
     except urllib.error.URLError as error:
         raise RuntimeError(f"Unable to fetch contributor data.") from error
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)